python/dgl/graphbolt/subgraph_sampler.py

```diff
@@ -0,0 +1,34 @@
+"""Subgraph samplers"""
+
+from torchdata.datapipes.iter import IterDataPipe
+
+
+class SubgraphSampler(IterDataPipe):
+    """A subgraph sampler.
+
+    It is an iterator equivalent to the following:
+
+    .. code:: python
+
+       for data in dp:
+           yield sampler_func(graph, data)
+
+    Parameters
+    ----------
+    dp : DataPipe
+        The datapipe.
+    graph : GraphStorage
```

Comment:
the sampler_func should already have the information of the graph, we don't need to pass the graph to the graph as a parameter to it.